bloom-housing/bloom
# refactor: remove backend dependencies from sidebar application components

# Pull Request Template

## Issue Overview

This PR addresses #2674 


## Description

Removes the backend dependencies from the LeasingAgent, GetApplication, and SubmitApplication components in the sidebar, and cleans up some tech debt along the way.

## How Can This Be Tested/Reviewed?

The three application sidebar sections (LeasingAgent, GetApplication, and SubmitApplication) should look the same as they did before with no functional changes, just a refactor.

[Contact Stories](https://deploy-preview-2675--dev-storybook-bloom.netlify.app/?path=/story/listing-sidebar-contact--no-address)
[ContactAddress Stories](https://deploy-preview-2675--dev-storybook-bloom.netlify.app/?path=/story/listing-sidebar-contact-address--default)

Notes:
* `LeasingAgent` component renamed to `Contact`, so that it can be generalized to show the address/contact information for anyone.
* `SidebarAddress` component renamed to `ContactAddress` so that it can be generalized to exist outside of the sidebar.
* Moved the `address` file from helpers into to two different components, `OneLineAddress` and `MultiLineAddress` to be able to have stories for them.
* Moved the function generating our postmark string into shared-helpers so that we can move that business logic out of the component and still have functional unit tests
* New header style added to the `Heading` component for the uppercase gray style, and am using that now in place of that classname where we have those headers
* Generally added lots of new stories and tests

This will be a breaking change for these components due to the new prop sets that no longer take in our `Listing` object.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
